syntax = "proto3";

package syft.grid.service;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";

message SignalingAnswerMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string payload = 3;
  string target_metadata = 4;
  syft.core.io.Address reply_to = 5;
}

message SignalingOfferMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string payload = 3;
  string target_metadata = 4;
  syft.core.io.Address reply_to = 5;
}

message OfferPullRequestMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}

message AnswerPullRequestMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}
